### ISSUE TITLE
[Feat] Improve SSIM loss 

### ISSUE THREAD

**Issue by isabelleAI** [kornia/kornia#473]
The SSIM loss function currently returns 
`loss = torch.clamp(-ssim_map + 1., min=0, max=1) / 2.`

but unless I'm mistaken I think it should be 
`loss = torch.clamp((-ssim_map + 1.)/2, min=0, max=1)`

**Comment by edgarriba** [comment]
@isabelleAI thanks for the feedback. You are right, with the current implementation I guess we are missing some gradients by clipping before averaging.

Besides, now that I'm looking at the actual implementation, probably we should rename the function to `sssim_loss` since it's actually computing the Structural Dissimilarity. This will change the API though, but I think for good. Any other thoughts ? /cc @ducha-aiki

**Comment by isabelleAI** [comment]
I agree on the renaming.  I went to check the source code precisely because I wasn't getting the results I expected for the SSIM metric, and saw it was returning the dissimilarity instead.

**Comment by v-prgmr** [comment]
Hi all,
I have stumbled upon another issue while using SSIM, using PyTorch's anomaly detection has pointed out that NAN is generated due to the division operation in line nr: 134 of [ssim.py](https://github.com/kornia/kornia/blob/master/kornia/losses/ssim.py)

`ssim_map = ((2. * mu1_mu2 + self.C1) * (2. * sigma12 + self.C2)) / \
            ((mu1_sq + mu2_sq + self.C1) * (sigma1_sq + sigma2_sq + self.C2))`

**Comment by edgarriba** [comment]
@v-prgmr can you share any code ?

**Comment by edgarriba** [comment]
fixed in #847
